pkg/storage/kubernetes.go

```diff
@@ -115,10 +115,16 @@ func (k kubernetesClient) ListFromCluster() (StorageList, error) {
 		return StorageList{}, errors.Wrapf(err, "unable to get PVC using selector %v", storagelabels.StorageLabel)
 	}
 
+	// to track volume mounts used by a PVC
+	validVolumeMounts := make(map[string]bool)
+
 	for _, pvc := range pvcs {
 		found := false
 		for _, volumeMount := range volumeMounts {
 			if volumeMount.Name == pvc.Name+"-vol" {
+				// this volume mount is used by a PVC
+				validVolumeMounts[volumeMount.Name] = true
+
 				found = true
```

Comment:
> I dont think you need this check at all, if you think about it; you can have a volume component in a devfile(and a pvc for it) and not be mounted on any container component. Is it really an err case?

Yes, we decided not to support such a case as this may lead to unintentional PVC usage.